cypress/e2e/shared/secrets.test.ts

```diff
@@ -0,0 +1,82 @@
+import {Organization} from '../../src/types'
+
+describe('Secrets', () => {
+  beforeEach(() => {
+    cy.flush()
+    cy.signin().then(() => {
+      cy.get('@org').then(({id}: Organization) =>
+        cy.fixture('routes').then(({orgs}) => {
+          cy.visit(`${orgs}/${id}/settings/`)
+          cy.getByTestID('tree-nav')
+          cy.setFeatureFlags({secretsUI: true}).then(() => {
+            cy.getByTestID('secrets--tab').click()
+          })
+        })
+      )
+    })
+  })
+
+  it('Secrets tab setup', () => {
+    // Empty state exists and displays context appropriate text
+    cy.getByTestID('empty-state').should('be.visible')
+    cy.getByTestID('empty-state').contains('Secrets')
+
+    // Create single secret, make sure it's visible
+    cy.get('@org').then(({id: orgID}: Organization) => {
+      cy.upsertSecret(orgID, {toEverybody: 'rupees baby'})
+        .then(() => {
+          cy.reload()
+          cy.getByTestID('secret-card--toEverybody').should('be.visible')
```

Comment:
can we check for the notification here?

Reply:
The upsert being used on line 26 is a straight API call to push the secret in - there isn't a frontend way to add a new secret yet.  So there isn't a notification that a new secret has been added it just shows up there as it now exists in the backend.

Reply:
Gotcha, my bad